core/jobs/batch_jobs/manual_voice_artist_name_job_test.py

```diff
@@ -0,0 +1,836 @@
+                        'duration_secs': 2
+                    }
+                }
+                }
+            }
+        change_list = [exp_domain.ExplorationChange({
+            'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'property_name': (
+                exp_domain.STATE_PROPERTY_RECORDED_VOICEOVERS),
+            'state_name': feconf.DEFAULT_INIT_STATE_NAME,
+            'new_value': new_voiceovers_dict,
+            'old_value': old_voiceover_dict
+        })]
+        exp_services.update_exploration(
+            self.editor_id_2, self.CURATED_EXPLORATION_ID_4,
+            change_list, 'Translation commits')
+
+        story_services.update_story(
+            self.owner_id, self.STORY_ID_2, [story_domain.StoryChange({
+                'cmd': 'add_story_node',
+                'node_id': 'node_2',
+                'title': 'Node2',
+            }), story_domain.StoryChange({
+                'cmd': 'update_story_node_property',
+                'property_name': 'exploration_id',
+                'node_id': 'node_2',
+                'old_value': None,
+                'new_value': self.CURATED_EXPLORATION_ID_4
+            })], 'Changes.')
+
+    def _create_non_curated_exploration(self) -> None:
+        """The method generates a non curated exploration."""
+        exploration_3 = self.save_new_valid_exploration(
+            self.NON_CURATED_EXPLORATION_ID_3,
+            self.owner_id,
+            title='title1',
+            category=constants.ALL_CATEGORIES[0],
+            end_state_name='End State',
+        )
+        self.publish_exploration(self.owner_id, exploration_3.id)
+
+
+class CreateExplorationVoiceArtistLinkModelsJobTests(
+    VoiceArtistMetadataModelsTestsBaseClass):
+
+    JOB_CLASS: Type[
+        manual_voice_artist_name_job.CreateExplorationVoiceArtistLinkModelsJob
+    ] = (
+        manual_voice_artist_name_job.CreateExplorationVoiceArtistLinkModelsJob
+    )
+
+    def test_empty_storage(self) -> None:
+        self.assert_job_output_is_empty()
+
+    def test_version_is_added_after_running_job(self) -> None:
+        self._create_curated_explorations()
+        self._create_non_curated_exploration()
+
+        self.assert_job_output_is([
+            job_run_result.JobRunResult(
+                stdout='Generated exploration voice artist link for %s.' %
+                self.CURATED_EXPLORATION_ID_1, stderr=''),
+            job_run_result.JobRunResult(
+                stdout='Generated exploration voice artist link for %s.' %
+                self.CURATED_EXPLORATION_ID_2, stderr=''),
+            job_run_result.JobRunResult(
+                stdout='Generated exploration voice artist link for %s.' %
+                self.CURATED_EXPLORATION_ID_4, stderr='')
+        ])
+
+        total_exploration_voice_artist_link_models = len(
+            voiceover_models.ExplorationVoiceArtistsLinkModel.get_all().fetch())
+
+        self.assertEqual(total_exploration_voice_artist_link_models, 3)
```

Comment:
You have no tests that actually test that the generated models of the job are correct...? Isn't that a key part of the behaviour that we need to verify?

Reply:
Added